flake.nix

```diff
@@ -0,0 +1,66 @@
+{
+  description = "heads flake, mostly for devshell for now";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.systems.follows = "systems";
+    };
+    systems.url = "github:nix-systems/default";
+  };
+
+  outputs = {
+    self,
+    flake-utils,
+    nixpkgs,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = nixpkgs.legacyPackages.${system};
+    in {
+      devShell = pkgs.mkShellNoCC {
```

Comment:
I recommend extracting this into e.g. a `devShell` attribute of `default.nix` and importing it from here:

```nix
# default.nix
let
  pkgs = <extract the pinned Nixpkgs from flake.lock here>;
in
{ pkgs ? pkgs, system ? builtins.currentSystem }:
{
  devShell = pkgs.mkShellNoCC {
  # ...
  };
}
```

```nix
# shell.nix
(import ./. {}).devShell
```

The `shell.nix` is optional for use with `nix-shell` as long as there is exactly one attribute in `default.nix` that evaluates  to a derivation.

```nix
# flake.nix
  # ...
  inherit (import ./. { inherit pkgs system; }) devShell; # or something along these lines
  # ...
```

This keeps compatibility with stable Nix, which without any doubt is here to stay. It also has the additional advantage that it avoids bloating your `flake.nix` for no good reason, and allows much easier re-use of Nix expressions, the number of which you may grow in the future.


An alternative approach is to fill `default.nix` with [`flake-compat` code](https://github.com/nix-community/flake-compat). (But IMO this just adds more dependencies and provides no material advantage - you'll still have a messy and growing `flake.nix` or will have to add even more files to keep things readable.)